### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ndorfin**
Dependencies:

- NPM
- Gulp

Uses:

- ES5
- ESLint
- Editorconfig
- `window` object name-spacing
- Function Objects and the `new` keyword to create instances
- 5 identified Shopify Form patterns

Includes:

- Rudimentary test cases. Run a python webserver from the root directory, then access `/test/` to see the test forms.

To do:

- [ ] Test in RequireJS / Shopify Plus environments
- [ ] Clean up references to Shopify Plus
- [x] Remove old `addressfinder.js`
- [x] Rename production output to `boot.js`

Nice to haves:

- [ ] Extend with user-defined field mappings?
- [ ] Unit testing?
- [ ] Integration testing?

### CODE REVIEW

File: src/_countryMode.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        title: 'Australia',
+        provinceAPIMappings: {
+          'ACT': ['Australian Capital Territory', 'ACT'],
+          'NSW': ['New South Wales', 'NSW'],
+          'NT' : ['Northern Territory', 'NT'],
+          'QLD': ['Queensland', 'QLD'],
+          'SA' : ['South Australia', 'SA'],
+          'TAS': ['Tasmania', 'TAS'],
+          'VIC': ['Victoria', 'VIC'],
+          'WA' : ['Western Australia', 'WA']
```
Comment (nigelramsay) [inline src/_countryMode.js]: there's an 'OT' or 'Other Territories'. For example:

3 Club Road, CHRISTMAS ISLAND OT 6798

File: src/_countryMode.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+          'Marlborough Region':       ['Marlborough', 'MBH'],
+          'Nelson Region':            ['Nelson', 'NSN'],
+          'Northland Region':         ['Northland', 'NTL'],
+          'Otago Region':             ['Otago', 'OTA'],
+          'Southland Region':         ['Southland', 'STL'],
+          'Taranaki Region':          ['Taranaki', 'TKI'],
+          'Tasman Region':            ['Tasman', 'TAS'],
+          'Waikato Region':           ['Waikato', 'WKO'],
+          'Wellington Region':        ['Wellington', 'WGN'],
+          'West Coast Region':        ['West Coast', 'WTC']
```
Comment (nigelramsay) [inline src/_countryMode.js]: There's a "No Region" - mostly for the Chatham Islands.

See: 10 Hotel Road, Chatham Island 8016

File: src/_loadAF.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    d.body.appendChild(s);
+  }
+
+  function _requireLoad(f) {
+    var af = w.require.config({
+      context: 'af',
+      baseUrl: '',
+      paths: {
+        addressfinder: 'https://api.addressfinder.io/assets/v3/core',
+        reqwest:       'https://files-abletech-nz.s3.amazonaws.com/addressfinder/reqwest',
```
Comment (nigelramsay) [inline src/_loadAF.js]: these dependencies might be better hosted on `api.addressfinder.io` as that `files-abletech-nz` container isn't really that permanent.

File: src/_countryMode.js (hunk 2 above)
Comment (ndorfin) [inline src/_countryMode.js]: @nigelramsay That region option won't have a Shopify mapping though. What do we do in that case?

File: src/_countryMode.js (hunk 1 above)
Comment (ndorfin) [inline src/_countryMode.js]: @nigelramsay That region option won't have a Shopify mapping though. What do we do in that case?

File: src/_loadAF.js (hunk 3 above)
Comment (ndorfin) [inline src/_loadAF.js]: Yup - we need to expose those in the production API environment.  I don't think they exist at the moment?

The more I think about RequireJS, the more I think we should just remove it. I've yet to find a working shopify instance that uses RequireJS. Powerplanter AU uses it, but it is broken.

**Pull request merged**